### PR TITLE
[EAG2 - 59] Enroll to game

### PR DESCRIPTION
**Pull request by RayWayy**

### CODE REVIEW

File: src/main/java/com/eleks/academy/whoami/core/state/WaitingForPlayers.java [hunk 1]
```diff
@@ -42,4 +43,15 @@ public int getPlayersInGame() {
 	public GameStatus getStatus() {
 		return WAITING_FOR_PLAYERS;
 	}
+
+	@Override
+	public List<SynchronousPlayer> getPlayersList() {
```
Comment (VolodymyrKr-Eleks) [inline src/main/java/com/eleks/academy/whoami/core/state/WaitingForPlayers.java]: Much better is to have a getter that just returns `Map<String, SynchronousPlayer> players` and in a place where it is used to have code that transforms it into needed value. So just have a getter and remove this method from interface

File: src/test/java/com/eleks/academy/whoami/controller/GameControllerTest.java [hunk 2]
```diff
@@ -148,4 +144,27 @@ void findByIdNoGameTest() throws Exception {
...
+		var expectedResponse = "{\"name\":\"newPlayer\",\"character\":null,\"question\":null,\"guess\":null}";
+
+		when(gameService.enrollToGame(id, newPlayer)).thenReturn(player1);
+
+		this.mockMvc.perform(
+						MockMvcRequestBuilders.post("/games/{id}/players", id)
+								.header("X-Player", newPlayer)
+								.contentType(MediaType.APPLICATION_JSON))
+				.andExpect(status().isOk())
+				.andExpect(content().string(expectedResponse));
```
Comment (VolodymyrKr-Eleks) [inline src/test/java/com/eleks/academy/whoami/controller/GameControllerTest.java]: compare please as JSON not as string

**Review by VolodymyrKr-Eleks — CHANGES_REQUESTED** [review]

File: src/test/java/com/eleks/academy/whoami/controller/GameControllerTest.java (hunk 2 above)
Comment (RayWayy) [reply src/test/java/com/eleks/academy/whoami/controller/GameControllerTest.java]: Resolved

File: src/main/java/com/eleks/academy/whoami/core/state/WaitingForPlayers.java (hunk 1 above)
Comment (RayWayy) [reply src/main/java/com/eleks/academy/whoami/core/state/WaitingForPlayers.java]: Resolved

**Review by VolodymyrKr-Eleks — APPROVED** [review]

**Pull request merged**